### PR TITLE
CI build now always uses Java11 as JAVA_HOME

### PR DESCRIPTION
**Pull request by asodja**
… so it works with studio-plugin project

### CODE REVIEW

**Review by wolfs — APPROVED** [review]
The changes look good. I suppose we set the toolchain for compilation in each of the subprojects, right? So we don't depend on the Java home there and we only depend on the Java home for selecting the Gradle version for running the build.

**Comment by asodja** [comment]
Yes, for a compilation we have it on every subproject.

File: .teamcity/settings/GradleProfilerTest.kt [hunk 1]
```diff
@@ -10,14 +10,20 @@ open class GradleProfilerTest(os: Os, javaVersion: JavaVersion) : BuildType({
...
     steps {
         gradle {
             tasks = "clean :test"
             buildFile = ""
-            gradleParams = "-s --build-cache ${toolchainConfiguration(os)}"
+            gradleParams = "-s" +
+                " --build-cache ${toolchainConfiguration(os)}" +
+                " -PtestJavaVersion=${javaVersion.majorVersion}" +
+                " -PtestJavaVendor=${javaVersion.vendor}"
+            param("org.jfrog.artifactory.selectedDeployableServer.defaultModuleVersionConfiguration", "GLOBAL")
```
Comment (wolfs) [inline .teamcity/settings/GradleProfilerTest.kt:26]: Why is this parameter here twice now?

File: build.gradle.kts [hunk 2]
```diff
@@ -99,10 +99,24 @@ tasks.processResources {
...
+        providers.gradleProperty("testJavaVendor").map {
+            when (it.toLowerCase()) {
+                "oracle" -> vendor.set(JvmVendorSpec.ORACLE)
+                "openjdk" -> vendor.set(JvmVendorSpec.ADOPTOPENJDK)
+            }
+        }.getOrNull()
+    }
+    javaLauncher.set(launcher)
+    // So processes started from test also use same Java version
+    environment = mapOf("JAVA_HOME" to javaLauncher.get().metadata.installationPath)
```
Comment (wolfs) [inline build.gradle.kts:119]: Would it be better here to only add the one extra environment variable instead of setting all of them?

**Review by wolfs — COMMENTED** [review]
I left some comments.

File: build.gradle.kts (hunk 2 above)
Comment (asodja) [reply build.gradle.kts]: Fixed

File: .teamcity/settings/GradleProfilerTest.kt (hunk 1 above)
Comment (asodja) [reply .teamcity/settings/GradleProfilerTest.kt:26]: Good catch, removed

**Review by wolfs — APPROVED** [review]
LGTM!

**Pull request merged**